### PR TITLE
Add nvim and xmonad configs

### PR DESCRIPTION
**Pull request by EnriqueVidal**
Adds a couple of new configs missing from the main branch.

- [x] Add cava
- [x] Add dunst
- [x] Add mpd
- [x] Add nvim config
- [x] Add xmonad config
- [x] Add xmobar config
- [x] Add READMEs
- [x] Make XDG compliant
- [x] Update Dockerfile

### CODE REVIEW

File: nvim/lua/lsp-config/language-servers.lua [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        -- Make the server aware of Neovim runtime files
+        library = vim.api.nvim_get_runtime_file("", true),
+      },
+      -- Do not send telemetry data containing a randomized but unique identifier
+      telemetry = {
+        enable = false,
+      },
+    },
+  },
+}
```
Comment (aalvarado) [inline nvim/lua/lsp-config/language-servers.lua:94]: lots of manual work here

File: nvim/lua/treesitter-config/init.lua [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  highlight = {
+    -- `false` will disable the whole extension
+    enable = true,
+
+    -- NOTE: these are the names of the parsers and not the filetype. (for example if you want to
+    -- disable highlighting for the `tex` filetype, you need to include `latex` in this list as this is
+    -- the name of the parser)
+    -- list of language that will be disabled
+    disable = { "c", "markdown", "rust" },
```
Comment (aalvarado) [inline nvim/lua/treesitter-config/init.lua:22]: do you know why these need to be disabled?

File: nvim/plugin/packer_compiled.lua [hunk 3]
```diff
@@ -0,0 +1,162 @@
+-- Automatically generated packer.nvim plugin loader code
```
Comment (aalvarado) [inline nvim/plugin/packer_compiled.lua:1]: if it's auto generated maybe it's safe to ignore

**Review by aalvarado — APPROVED** [review]

File: nvim/plugin/packer_compiled.lua (hunk 3 above)
Comment (EnriqueVidal) [reply nvim/plugin/packer_compiled.lua:1]: Yeah it was an oversight will ignore

File: nvim/lua/treesitter-config/init.lua (hunk 2 above)
Comment (EnriqueVidal) [reply nvim/lua/treesitter-config/init.lua:22]: Doesn't need to, I just didn't bother configuring those I will comment this line

File: nvim/lua/lsp-config/language-servers.lua (hunk 1 above)
Comment (EnriqueVidal) [reply nvim/lua/lsp-config/language-servers.lua]: yeah I sorta wish it was closer to coc but sadly the choice is either speed or convenience :(

File: .config/bash/environments [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+export XDG_CONFIG_HOME="$HOME/.config"
+export XDG_STATE_HOME="$HOME/.local/state"
+export XDG_CACHE_HOME="$HOME/.cache"
+export XDG_RUNTIME_DIR="/run/user/$UID"
+
+export ACKRC="$XDG_CONFIG_HOME/ack/ackrc"
+export PSQLRC="$XDG_CONFIG_HOME/pg/psqlrc"
+export XINITRC="$XDG_CONFIG_HOME/X11/xinitrc"
+
+# vim: set ft=sh:
```
Comment (aalvarado) [inline .config/bash/environments:28]: perhaps the env var for setting qt theme settings would be useful here

File: .config/nvim/lua/plugins/packer.lua [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+	-- Themes
+	use("EdenEast/nightfox.nvim")
+	use({
+		"nvim-lualine/lualine.nvim",
+		requires = { "kyazdani42/nvim-web-devicons", opt = true },
+	})
+
+	-- LSP
+	use("neovim/nvim-lspconfig") -- Collection of configuration for LSP
+	use("williamboman/mason.nvim") -- Lagunage Server Auto installation
```
Comment (aalvarado) [inline .config/nvim/lua/plugins/packer.lua:27]: typo

File: .config/vim/vimrc [hunk 6]
```diff
@@ -8,59 +38,33 @@ filetype off
...
-Plug 'HerringtonDarkholme/yats.vim'
 Plug 'vim-scripts/nginx.vim'
-Plug 'vim-scripts/AutoTag'
 Plug 'vim-scripts/haskell.vim'
 Plug 'vim-scripts/LargeFile'
 Plug 'mattn/emmet-vim'
-Plug 'pangloss/vim-javascript'
-Plug 'docunext/closetag.vim'
-Plug 'nono/vim-handlebars'
-Plug 'briancollins/vim-jst'
```
Comment (aalvarado) [inline .config/vim/vimrc:34]: if you add polyglot you can remove all the syntax plugins

File: .config/vim/vimrc [hunk 7]
```diff
@@ -75,7 +79,7 @@ if filereadable(expand("~/.vim/before"))
 endif
 
 set hidden
-set number
+set relativenumber
```
Comment (aalvarado) [inline .config/vim/vimrc:82]: you're team relativenumber now :D

**Review by aalvarado — APPROVED** [review]

**Review by aalvarado — APPROVED** [review]

File: .config/vim/vimrc (hunk 6 above)
Comment (EnriqueVidal) [reply .config/vim/vimrc:34]: can you link  me?

File: .config/nvim/lua/plugins/packer.lua (hunk 5 above)
Comment (EnriqueVidal) [reply .config/nvim/lua/plugins/packer.lua:27]: that's the cool way to spell it, will update

File: .config/bash/environments (hunk 4 above)
Comment (EnriqueVidal) [reply .config/bash/environments:28]: What variable is that? and what's the most sensible default?

File: .config/vim/vimrc (hunk 6 above)
Comment (aalvarado) [reply .config/vim/vimrc:34]: https://github.com/sheerun/vim-polyglot

File: .config/bash/environments (hunk 4 above)
Comment (aalvarado) [reply .config/bash/environments:28]: `QT_STYLE_OVERRIDE`
sensible default would be `gtk2`

https://wiki.archlinux.org/title/Qt#Appearance

or 

`QT_QPA_PLATFORMTHEME=qt5ct` but needs qt5ct installed

Comment (EnriqueVidal) [reply .config/bash/environments:28]: we can set both, the second could be done by:

```bash
QT5CT_PATH=`which qt5ct`

if [[ -x $QT5CT_PATH ]]; then
  export QT_QPA_PLATFORMTHEME=qt5ct
fi
```

**Pull request merged**